File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> indices;
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); i++) {
            int difference = target - nums[i];
            if ( seen.find(difference) != seen.end() ) {
                // existed
                indices.push_back(seen[difference]);
                indices.push_back(i);
                return indices;
            } else {
                seen[nums[i]] = i;
            }
        }

        return indices;
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-0.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1:
            if n in seen:
                return False
            seen.add(n)
            n = sum([(ord(c)-ord('0'))**2 for c in str(n)])

        return True
            